src/main/java/com/powsybl/sld/server/SingleLineDiagramService.java

```diff
@@ -73,23 +74,27 @@ Pair<String, String> generateSvgAndMetadata(UUID networkUuid, String variantId,
             throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Voltage level or substation " + id + " not found");
         }
 
-        try (StringWriter svgWriter = new StringWriter();
-             StringWriter metadataWriter = new StringWriter()) {
-            LayoutParameters layoutParameters = new LayoutParameters(LAYOUT_PARAMETERS);
-            layoutParameters.setLabelCentered(diagParams.isLabelCentered());
-            layoutParameters.setLabelDiagonal(diagParams.isDiagonalLabel());
-            layoutParameters.setUseName(diagParams.isUseName());
-            layoutParameters.setAddNodesInfos(true); // only used for voltage level diagrams
+        try (var svgWriter = new StringWriter();
+             var metadataWriter = new StringWriter()) {
 
             ComponentLibrary compLibrary = ComponentLibrary.find(diagParams.getComponentLibrary())
                     .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Component library '" + diagParams.getComponentLibrary() + "' not found"));
 
             var defaultDiagramStyleProvider = diagParams.isTopologicalColoring() ? new TopologicalStyleProvider(network)
                                                                                  : new NominalVoltageDiagramStyleProvider(network);
-            var labelProvider = new DefaultDiagramLabelProvider(network, compLibrary, layoutParameters);
+            LayoutParameters layoutParameters = null;
+            DefaultDiagramLabelProvider labelProvider = null;
+            if (diagParams.getSldDisplayMode() == SldDisplayMode.FEEDER_POSITION) {
+                layoutParameters = fillLayoutParameters(diagParams, false, true);
```

Comment:
Is not a clear design as at first view, we don't known what is first and second boolean.
```java
layoutParameters = ...
if (diagParams.getSldDisplayMode() == SldDisplayMode.FEEDER_POSITION) {
    layoutParameters.setAddNodesInfos(...)
   ....
} ...
```